lib/bootstrap/bundler.rb

```diff
@@ -128,6 +127,11 @@ def invoke!(options = {})
       ::Bundler.settings.set_local(:gemfile, LogStash::Environment::GEMFILE_PATH)
       ::Bundler.settings.set_local(:without, options[:without])
       ::Bundler.settings.set_local(:force, options[:force])
+      # This env setting avoids the warning given when bundler is run as root, as is required
+      # to update plugins when logstash is run as a service
+      # Note: Using an `ENV` here, as ::Bundler.settings.set_local(:silence_root_warning, true)
+      # does not work (set_global *does*, but that seems too drastic a change)
+      ENV["BUNDLE_SILENCE_ROOT_WARNING"] = "true"
```

Comment:
Truly a nitpick, since we exit the process before doing anything else that will be affected by this, but it may be good to clean up after ourselves so we don't pollute the ENV for longer than we need to.

~~~ ruby
def with_env(modifications)
  backup_env = ENV.to_hash
  ENV.replace(backup_env.merge(modifications))

  yield
ensure
  ENV.replace(backup_env)
end
~~~

Which would allow us to do:

~~~ ruby
  with_env("BUNDLE_SILENCE_ROOT_WARNING" => "true") do
    if !debug?
        # Will deal with transient network errors
        execute_bundler_with_retry(options)
    else
      options[:verbose] = true
      execute_bundler(options)
    end
  end

Reply:
Yeah, I like that. Not a fan of unintended side effects. Even if it's not a problem now, its the kind of subtle thing that can bite later